### ISSUE TITLE
Version 1.1.1: 500 Error When Installing Certain Extensions

### ISSUE THREAD

**Issue by pybokeh** [jupyterlab/jupyterlab#7133]
## Description

When attempting to install certain extensions, I would get: Error: 500 (Internal Server Error)
**NOTE:** I can install other extensions fine, but on a few, I would get 500 error.

## Reproduce

I get error when attempting to install the snippets extension: @quentinandre/jupyterlab_snippets

and also a few other ones.

## Expected behavior

The snippets extension gets installed without error

## Context
On Windows 10 machine with no admin rights.
Fresh new conda environment with JupyterLab version 1.1.1 per conda list output:
```
(pydata64) λ conda list | find "jupyter"
jupyter                   1.0.0                      py_2    conda-forge
jupyter-contrib-core      0.3.3                    pypi_0    pypi
jupyter-contrib-nbextensions 0.5.1                    pypi_0    pypi
jupyter-highlight-selected-word 0.2.0                    pypi_0    pypi
jupyter-latex-envs        1.4.6                    pypi_0    pypi
jupyter-nbextensions-configurator 0.4.1                    pypi_0    pypi
jupyter_client            5.3.1                      py_0    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyter_server            0.1.1                    py37_0    conda-forge
jupyterlab                1.1.1                      py_0    conda-forge
jupyterlab_pygments       0.1.0                      py_0    conda-forge
jupyterlab_server         1.0.6                      py_0    conda-forge
```

Output of my console when error occurs:
```
[W 07:55:18.683 LabApp] 500 POST /jupyter/lab/api/extensions?1567166112596 (::1): '@jupyterlab/apputils'
[W 07:55:18.685 LabApp] '@jupyterlab/apputils'
[E 07:55:18.686 LabApp] {
      "Host": "localhost:8888",
      "Connection": "keep-alive",
      "Content-Length": "70",
      "Sec-Fetch-Mode": "cors",
      "Origin": "http://localhost:8888",
      "X-Xsrftoken": "2|bd469540|fb320d6843309d147edf9364c5fb3a51|1566940851",
      "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36",
      "Content-Type": "application/json",
      "Accept": "*/*",
      "Sec-Fetch-Site": "same-origin",
      "Referer": "http://localhost:8888/jupyter/lab?",
      "Accept-Encoding": "gzip, deflate, br",
      "Accept-Language": "en-US,en;q=0.9",
      "Cookie": "username-localhost-8888=\"2|1:0|10:1566923535|23:username-localhost-8888|44:NjBjODNlZmRhZjY5NGQyYzkzYmJkZmI0ODM3NzZmYTk=|c0b9624506851e541c2f7e38674ccea10d89b14879ca43bed9f26654cefb3e67\"; _xsrf=2|bd469540|fb320d6843309d147edf9364c5fb3a51|1566940851"
    }
[E 07:55:18.697 LabApp] 500 POST /jupyter/lab/api/extensions?1567166112596 (::1) 6084.37ms referer=http://localhost:8888/jupyter/lab?
```

**Comment by jasongrout** [comment]
For completeness, can you list the other extensions that are having the same problem?

Did `@quentinandre/jupyterlab_snippets` work in 1.0? When I go to the npm page for `@quentinandre/jupyterlab_snippets`, it appears that the repo is not on github anymore.

**Comment by jasongrout** [comment]
And for that particular extension, it appears to have a server component and a js component. It appears that there may be an error in the server component of that extension (i.e., not a jlab issue?)

**Comment by furkantektas** [comment]
I've also experienced the same issue with `@ryantam626/jupyterlab_black` extension. 

```
jupyter_client            5.3.1                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterlab                1.1.1                      py_0    conda-forge
jupyterlab_server         1.0.6                      py_0    conda-forge
```

This problem can also be reproducable using the CLI.

```
(conda_env) ➜  jupyter labextension install jupyterlab_formatblack
An error occured.
KeyError: '@jupyterlab/notebook'
See the log file for details:  /tmp/jupyterlab-debug-v0uqocrz.log
```

**Comment by blink1073** [comment]
Ah, the issue here is that we had a bug that was obscuring the real message.  The fix is in #7165.

After the fix:

```
Conflicting Dependencies:
JupyterLab              Extension      Package
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/application
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/apputils
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/notebook

An error occured.
ValueError: This extension does not yet support the current version of JupyterLab.


Conflicting Dependencies:
JupyterLab              Extension      Package
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/application
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/apputils
>=1.1.0 <1.2.0          >=0.16.0 <0.20.0@jupyterlab/notebook
See the log file for details:  /tmp/jupyterlab-debug-qcgg22sv.log
Exiting application: jupyter
```

**Comment by jasongrout** [comment]
Fixed in #7165, according to Steve's comment.
